documentation/helpers/ArgTypes/index.tsx

```diff
@@ -20,15 +20,41 @@ const ComponentDescription = ({ name, children }: { name: string; children: Reac
   )
 }
 
+function useWindowWidth() {
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth)
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowWidth(window.innerWidth)
+    }
+
+    window.addEventListener('resize', handleResize)
+
+    return () => {
+      window.removeEventListener('resize', handleResize)
+    }
+  }, [])
+
+  return windowWidth
+}
+
 export const ArgTypes = <T extends FC>({ of, description, subcomponents = null }: Props<T>) => {
+  const windowWidth = useWindowWidth()
+
   if (!subcomponents) return <StorybookArgTypes of={of} />
 
   const { displayName: name = 'Root' } = of // "Root" in case the root component is missing a displayName
   const subComponentsList = Object.entries(subcomponents)
 
+  const tabsOrientation = windowWidth < 640 ? 'horizontal' : 'vertical'
```

Comment:
Could we just create a `useWindowOrientation` hook just storing `horizontal | vertical` for do not rerender everytime the width changes when storing the with on a react state?